### PR TITLE
Add friendlier error for two adjacent elements

### LINKED ISSUE #1584: Parse error is confusing when trying to return two elements
(Opening here because esprima-fb apparently doesn't accept issues.)

If you write

```
return <div className="one" /><div className="two" />;
```

then you get "Unexpected identifier" because it tries to parse it as

```
return React.DOM.div({className: "one"})  < div className ...
```

That is, the `<` of the second div is interpreted as a less than. I think we can give nicer errors here if we disallow the less than operator directly after an XJSElement. I don't think this prevents any real uses but if it did, you could always wrap it in parens:

```
return (<div className="one" />) < threshold;
```

cc @jeffmo @syranide

### PR DESCRIPTION
**Pull request by sophiebits**
Fixes facebook/react#1584. We've had a handful of people confused about this recently so I think this would be nice to do if it doesn't conflict with any real uses that you can think of.

Test Plan: npm test

cc @jeffmo @syranide

### CODE REVIEW

**Comment by syranide** [comment]
As JSX is just sugar for function-calls, it does technically prevent _valid_ uses. However, JSX is _not_ intended as a substitute for function-calls so comparing the return-value of a JSX-tag never makes sense. If someone for some superweird reason would want that, they'll just have to assign it a variable first and compare that (or possibly flip the expression).

:+1:

**Comment by sophiebits** [comment]
Or parenthesize it, as I verified works in a test case.

**Comment by syranide** [comment]
Ah yes ofc.

File: esprima.js [hunk 1]
```diff
@@ -5648,6 +5649,19 @@ parseYieldExpression: true
...
+        // When (erroneously) writing two adjacent tags like
+        //
+        //     var x = <div>one</div><div>two</div>;
+        //
+        // the default error message is a bit incomprehensible. Since it's
+        // rarely (never?) useful to write a less-than sign after an XJS
+        // element, we disallow it here in the parser in order to provide a
+        // better error message. (In the rare case that the less-than operator
+        // was intended, the left tag can be wrapped in parentheses.)
+        if (!origInXJSChild && lookahead.value === '<') {
```
Comment (jeffmo) [inline esprima.js]: s/`lookahead.value === '<'`/`match('<')`/

(this also checks that the type of the token is `Punctuator`)

Comment (sophiebits) [inline esprima.js]: Done.

**Comment by jeffmo** [comment]
love it!

**Pull request merged**